cookbooks/bcpc/libraries/utils.rb

```diff
@@ -22,6 +22,22 @@
 require 'thread'
 require 'ipaddr'
 
+# create databag if missing
+def create_databag(name)
+  if !Chef::DataBag.list.key?(name)
+    bag = Chef::DataBag.new
+    bag.name(name)
+    bag.create
+  end
+end
+
+# load chef vault item from vault bag
+def get_secret(bag,item)
+require 'chef-vault'
+  entry = ChefVault::Item.load(bag, item)
```

Comment:
Chef API will handle that, please check https://github.com/Nordstrom/chef-vault/blob/master/lib/chef-vault/item_keys.rb#L114-L127